dune/iga/geometrykernel/findintersection.hh

```diff
@@ -23,20 +23,19 @@ namespace Dune::IGANEW {
     const FieldMatrix<ScalarType, dim, 2> linearSystem({curveP1 - curveP0, -dir});
     const FieldVector<ScalarType, dim> b = pos - curveP0;
 
-    try {
+    if (not FloatCmp::eq(linearSystem.determinant(), 0.0, tol)) {
```

Comment:
Is there a reaseon why we shouldn't check for `det == 0` ? 
I don't really like the exception here, also it seems to fail when feexceptions are enabled

Reply:
Ah maybe i did a mistake ther, but using `DUNE_FMatrix_WITH_CHECKING` and solve actually checks the determinant inside. I didn't like the idea of computing the determinant twice. See https://gitlab.dune-project.org/core/dune-common/-/blob/a658c135ed61b66c7b87ec3c2e66966eba762c96/dune/common/densematrix.hh#L954-957

Reply:
But I'm afraid to maintain the preprocessor define DUNE_FMatrix_WITH_CHECKING is not worth it. We shopuld keep your refactoring for now. Even if we do unnecessary computations.